### PR TITLE
Story/vspc 82

### PR DESCRIPTION
**Pull request by Sagar-k93**
On slides: text blocks should allow markdown. So that the user can make words bold, italics, include links, etc. The markdown should be rendered a) on the public page b) on a preview version for the text block when editing.

### CODE REVIEW

**Comment by jdamerow** [comment]
please resolve conflicts and codefactor issues

File: vspace/src/main/webapp/WEB-INF/views/exhibition/module.jsp [hunk 1]
```diff
@@ -125,11 +135,9 @@ body {
...
                 </c:if>
                 <c:if
-                    test="${contents['class'].simpleName ==  'TextBlock'}">
-                    <div id="${contents.id}" class="textDiv">
-                        <p>${contents.text}</p>
-                    </div>
-                </c:if>
+                        test="${contents['class'].simpleName ==  'TextBlock'}">
+                        <div id="${contents.id}" class="textDiv">${contents.htmlRenderedText()}</div>
+                    </c:if>
```
Comment (prjadhav14) [inline vspace/src/main/webapp/WEB-INF/views/exhibition/module.jsp:140]: indentation with respect to opening if statement.

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp [hunk 2]
```diff
@@ -543,13 +547,16 @@ $(document).ready(function() {
     });
 
     $("#addImgAlert").draggable();
-    $("#addTextAlert").draggable();
+
+    //$("#editTextAlert").draggable();
+
```
Comment (prjadhav14) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.jsp:552]: remove if not required

File: vspace/pom.xml [hunk 3]
```diff
@@ -185,7 +185,14 @@
...
             <scope>runtime</scope>
         </dependency>
-
+        
+        <!-- MarkDown -->
+		<dependency>
+    		<groupId>com.atlassian.commonmark</groupId>
+    		<artifactId>commonmark</artifactId>
+    		<version>0.15.0</version>
+		</dependency>
```
Comment (jdamerow) [inline vspace/pom.xml:194]: indentation

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java [hunk 4]
```diff
@@ -155,4 +163,21 @@ public void deleteSlideById(String slideId, String moduleId) {
         }
         return sequenceSlides;
     }
+
+    @Override
+    public ISlide convertTextToMarkdown(ISlide slide) {
+        List<IContentBlock> content = slide.getContents();
+        for(int i=0; i<content.size(); ++i) {
+            if(content.get(i).getClass().equals(TextBlock.class)) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java:171]: does `instance of` not work here?

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java [hunk 5]
```diff
@@ -155,4 +163,21 @@ public void deleteSlideById(String slideId, String moduleId) {
         }
         return sequenceSlides;
     }
+
+    @Override
+    public ISlide convertTextToMarkdown(ISlide slide) {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java:168]: where is that being used?

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Pull request merged**